### ISSUE TITLE
IATI Publisher production log-in not working

### ISSUE THREAD

**Issue by emmajclegg** [younginnovations/iatipublisher#1580]
Users are getting an error message about their password when trying to log-in to IATI Publisher ("password should be string").

This is urgent to fix as it is preventing log-in (for all users as far as we can tell)

![image](https://github.com/user-attachments/assets/a5c3b5a9-347a-49d2-8623-bf58d8e855a0)

**Comment by emmajclegg** [comment]
Issue has been fixed - details of the issue and fix applied to be added here

**Comment by PG-Momik** [comment]
The "password must be string" issue is a known and documented bug in IATI Publisher. You can find the details in the "Understanding 'password must be string'" section of the attached document.


[Handover doc-Password should be string.pdf](https://github.com/user-attachments/files/17484867/Handover.doc-Password.should.be.string.pdf)

Since it has been a while since we last encountered the "Password should be string" error, we fell out of the habit of checking production after deployments.

**Comment by emmajclegg** [comment]
> Since it has been a while since we last encountered the "Password should be string" error, we fell out of the habit of checking production after deployments.

To be clear, if this is a known issue, we expect a more thorough QA process that avoids issues like this slipping undetected into production. It has such a damaging impact to users on confidence in IATI Publisher and takes up our time in support trying to troubleshoot. 

I'll leave this issue open for the time being, given it sounds like the issue is being revisited with your DevOps engineer.

**Comment by PG-Momik** [comment]
@emmajclegg  update on this issue:

The two possible solutions for this issue suggested by the Devops are:
1. Simplified Solution: Remove encryption from the frontend payload. This feature is redundant since is the primary cause of this issue.
2. Alternative Solution (More Complex and Time-Consuming):    Automatically clear the cache with each deployment. This would require frequent and repetitive actions and would also increase the build time for our CI/CD pipelines.

--- 

### After an internal discussion and reviewing the pros and cons we've decided to go with the Simplified solution.


#### Solution 1: Simplified Solution
- Involves removing the main cause of the problem by removing frontend encryption. This will remove our dependency on Github Cache.
- Requires us to test some number of builds (say 20-30 builds) to ensure that "password should be string" is actually removed. Assuming each build and deployment takes 15 mins, we're looking at 5 hrs or more.
 
 #### Solution 2: Alternative solution
- Involves not changing our current login flow. Doesnt remove our dependency from Github cache. 
- Basically implement an additional CI/CD workflow that clears the entire cache in each deployment. This will cause build times in increase. Will also increase our CI/CD steps and decrease efficiency.



@pravin-yipl Let me know if anything from the discussion is not mentioned in the summary.

---

cc: @praweshsth @BibhaT  @Yash-ftW

**Comment by pravin-yipl** [comment]
Everything looks alright. Please continue

**Comment by emmajclegg** [comment]
Go ahead with your suggested solution 1. Thanks for the update on this

**Comment by PG-Momik** [comment]
@emmajclegg  This has been deployed to staging. There will not be any visible changes. In oru 20 test runs we didn't face any 'Password should be string' issue or any problems in any of the user modules. (Register, login, update, etc)
